yt/frontends/amrvac/data_structures.py

```diff
@@ -245,56 +245,50 @@ def _set_code_unit_attributes(self):
-            unit_numberdensity = self.units_override.get('numberdensity_unit', (1, 'cm**-3'))
-            unit_velocity = self.units_override.get('velocity_unit', (0, 'cm*s**-1'))
-            unit_temperature = self.units_override.get('temperature_unit', (1, 'K'))
-            mylog.info('Overriding numberdensity_unit: {:1.0e} {}.'.format(unit_numberdensity[0], unit_numberdensity[1]))
+            length_override = self.units_override.get('length_unit', (1, 'cm'))
+            numberdensity_override = self.units_override.get('numberdensity_unit', (1, 'cm**-3'))
+            velocity_override = self.units_override.get('velocity_unit', (0, 'cm*s**-1'))
+            temperature_override = self.units_override.get('temperature_unit', (1, 'K'))
+            mylog.info('Overriding numberdensity_unit: {:1.0e} {}.'.format(numberdensity_override[0],
+                                                                           numberdensity_override[1]))
         else:
-            # Next, check for unit normalisations in the .dat file. Do not use .get() with default value added
-            # to explicitly print a warning to the console.
-            try:
-                unit_length = (self.parameters['unit_length'], 'cm')
-                unit_numberdensity = (self.parameters['unit_numberdensity'], 'cm**-3')
-                unit_temperature = (self.parameters['unit_temperature'], 'K')
-                unit_velocity = (self.parameters['unit_velocity'], 'cm*s**-1')
-            # If nothing present in datfile, use default values and print warning
-            except KeyError:
-                mylog.warning('No unit normalisations found in datfile and no overrides specified. Using defaults.')
-                unit_length = (1, 'cm')
-                unit_numberdensity = (1, 'cm**-3')
-                unit_temperature = (1, 'K')
-                unit_velocity = (0, 'cm*s**-1')
-
-        # Convert to YT quantities
-        unit_length = self.quan(unit_length[0], unit_length[1])
-        unit_numberdensity = self.quan(unit_numberdensity[0], unit_numberdensity[1])
-        unit_temperature = self.quan(unit_temperature[0], unit_temperature[1])
-        unit_velocity = self.quan(unit_velocity[0], unit_velocity[1])
-
-        He_abundance = 0.1  # hardcoded in AMRVAC
-
-        unit_density = (1.0 + 4.0*He_abundance) * mass_hydrogen_cgs * unit_numberdensity
-        if unit_velocity == 0:
-            unit_pressure = ((2.0 + 3.0*He_abundance) *
-                             unit_numberdensity * boltzmann_constant_cgs  * unit_temperature).to('dyn*cm**-2')
-            unit_velocity = (np.sqrt(unit_pressure / unit_density)).to('cm*s**-1')
+            # if nothing is specified, use the default AMRVAC normalisations
+            length_override = (1, 'cm')
+            numberdensity_override = (1, 'cm**-3')
+            temperature_override = (1, 'K')
+            velocity_override = (0, 'cm*s**-1')
```

Comment:
so the logic of this block is "as long as SOME override has been provided, we'll try and fetch 4 specific quantities in it, with defaults exactly identical to those we set if no override is provided"... This feels like duplicated code.
Unless I misunderstood, it is strictly equivalent to remove the `if` line and the `else` block. (except for the logging entry)

Reply:
aaaah you're right of course! stupid me 🤦‍♂️